bittensor/config.py

```diff
@@ -398,6 +483,99 @@ def __get_required_args_from_parser(parser: argparse.ArgumentParser) -> List[str
                 required_args.append(prefix + action.dest)
         return required_args
 
+    def load_config_from_env_vars(self):
+        """
+        Store the key-value pairs from environment variables starting with BTCLI_ in env_config.
+        The environment variable names are converted to nested dictionary keys.
+        """
+        env_vars = {k: v for k, v in os.environ.items() if k.startswith("BTCLI_")}
+        for var, value in env_vars.items():
+            key = var[6:].lower()
+            key = key.replace("_", ".")
+            self.env_config[key] = value
+
+    def load_or_create_generic_config(self):
+        config_path = os.path.expanduser(defaults.config.path)
+        config_file_yaml = os.path.join(config_path, "btcliconfig.yaml")
+        config_file_yml = os.path.join(config_path, "btcliconfig.yml")
+
+        config_file = None
+        if os.path.exists(config_file_yaml):
+            config_file = config_file_yaml
+        elif os.path.exists(config_file_yml):
+            config_file = config_file_yml
+
+        if config_file:
+            with open(config_file, "r") as file:
+                config_data = yaml.safe_load(file)
+        else:
+            config_data = defaults.toDict()
+            # Remove data we don't want to save in the default config, maybe
+            # it's better to use a second define for it and not defaults
+            config_data.pop("config", None)
+            config_data.pop("profile", None)
+            config_data.pop("wallet", None)
+            os.makedirs(config_path, exist_ok=True)
+            with open(config_file_yml, "w+") as file:
+                yaml.safe_dump(config_data, file)
+
+        def flatten_dict(d, parent_key=""):
+            for k, v in d.items():
+                new_key = f"{parent_key}.{k}" if parent_key else k
+                if isinstance(v, dict):
+                    flatten_dict(v, new_key)
+                else:
+                    self.generic_config[new_key] = v
+
+        if not config_data:
+            config_data = {}
+
+        flatten_dict(config_data)
+
+    def load_active_profile(self):
+
+        profile_name = (self.params_config.get('profile.active') or
```

Comment:
Would it be simpler to just build a merged config up to this point and read config.profile.active rather than all this switching?